src/main/resources/templates/c_plus_plus/test/testUtils/AbstractTest.py

```
@@ -0,0 +1,274 @@
+
+        if self.timeoutSec > 0:
+            # Run with timeout:
+            with self.__timeout(self.timeoutSec):
+                try:
+                    self._run()
+                except TestFailedError:
+                    printTester(f"'{self.name}' failed.")
+                except TimeoutError:
+                    self._timeout()
+                except Exception as e:
+                    self.__markAsFailed(f"'{self.name}' had an internal error. {str(e)}.\nPlease report this to an instructor!")
```

Ruff

Use explicit conversion flag

Replace with conversion flag

(RUF010)